source/shared/cpp/ObjectModel/TextBlock.cpp

```diff
@@ -165,3 +168,141 @@ void TextBlock::SetHorizontalAlignment(const HorizontalAlignment value)
     m_hAlignment = value;
 }
 
+std::string TextBlock::ParseISO8601UsingRegex() const
+{
+	std::regex pattern("\\{\\{(DATE|Date|TIME|Time)\\((\\d{4})-{1}(\\d{2})-{1}(\\d{2})T(\\d{2}):{1}(\\d{2}):{1}(\\d{2})(Z|([+-])(\\d{2}):{1}(\\d{2}))(\\)|,Short\\)|,SHORT\\)|,Long\\)|,LONG\\))\\}\\}");
+    std::smatch matches;
+    std::string text = m_text;
+    std::ostringstream parsedostr;
+    time_t offset = 0; 
+    bool isDate = false, isShort = true;
+    enum MatchIndex
+    {
+        IsDate = 1,
+        YEAR,
+        Month,
+        Day,
+        Hour,
+        Min,
+        Sec,
+        TimeZone,
+        TZHr = 10,
+        TZMn,
+        Format,
+    };
+    int factor = 1, hours = 0, minutes = 0;
+    struct tm parsedTm = { 0 };
+    int *addrs[] = {&parsedTm.tm_year, &parsedTm.tm_mon, 
+        &parsedTm.tm_mday, &parsedTm.tm_hour, &parsedTm.tm_min, 
+        &parsedTm.tm_sec, &hours, &minutes}; 
+    std::vector<int> indexer = {YEAR, Month, Day, Hour, Min, Sec, TZHr, TZMn};
+
+    while(std::regex_search(text, matches, pattern))
+    {
+        parsedostr << matches.prefix().str();
+
+        if(matches[IsDate].matched)
+        {
+            // Date is matched
+            isDate = (matches[IsDate].str()[0] == 'D')? true : false;
```

Comment:
>? true : false; [](start = 54, length = 15)

Aren't these unnecessary? matches[IsDate].str()[0] == 'D' is already a bool.

Reply:
do you mean matches[isDate] is bool? it would match both Date and Time string.


---
In reply to: [149192396](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149192396) [](ancestors = 149192396)

Reply:
I mean (matches[IsDate].str()[0] == 'D') is already a boolean expression. The conditional operator is not necessary

---
In reply to: [149215274](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149215274) [](ancestors = 149215274,149192396)

Reply:
thanks; that's good point

---
In reply to: [149217300](https://github.com/Microsoft/AdaptiveCards/pull/871#discussion_r149217300) [](ancestors = 149217300,149215274,149192396)